llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4550,11 +4777,21 @@ VectorizationFactor LoopVectorizationPlanner::selectVectorizationFactor() {
   }
 
   for (auto &P : VPlans) {
-    for (ElementCount VF : P->vectorFactors()) {
+    SmallVector<ElementCount, 1> VFs(P->vectorFactors());
+    auto RUs = ::calculateRegisterUsage(*P, VFs, TTI, CM.ValuesToIgnore);
+    for (auto [VF, RU] : zip_equal(VFs, RUs)) {
```

Comment:
nit: to avoid having to create a separate `SmallVector<> VFs`, I'd suggest doing:
```suggestion
    auto RUs = ::calculateRegisterUsage(*P, VFs, TTI, CM.ValuesToIgnore);
    for (auto [I, VF] : enumerate(VFs)) {
```
and use it as `RUs[I].MaxLocalUsers` below.

Reply:
I've updated the above comment, this should work.

Reply:
Done.